src/store/accounts/api.ts

```diff
@@ -0,0 +1,113 @@
+import { atomFamilyWithInfiniteQuery } from 'jotai-query-toolkit';
+import { atomFamilyWithQuery } from '@store/query';
+import { QueryRefreshRates } from '@common/constants';
+import { apiClientState } from '@store/common/api-clients';
+import type {
+  MempoolTransaction,
+  Transaction,
+  AccountDataResponse,
+} from '@stacks/stacks-blockchain-api-types';
+import type { AccountBalanceResponseBigNumber, AddressBalanceResponse, Keys } from './types';
+
+import BigNumber from 'bignumber.js';
+
+enum AccountClientKeys {
+  InfoClient = 'account/InfoClient',
+  BalancesClient = 'account/BalancesClient',
+  TransactionsClient = 'account/TransactionsClient',
+  MempoolTransactionsClient = 'account/MempoolTransactionsClient',
+}
+
+type PrincipalWithNetworkUrl = [principal: string, networkUrl: string];
+type PrincipalWithLimitNetworkUrl = [principal: string, limit: number, networkUrl: string];
+
+export interface ResultsWithLimitOffsetTotal<T> {
+  limit: number;
+  offset: number;
+  total: number;
+  results: T[];
+}
+
+const keys: Keys[] = [
+  'balance',
+  'total_sent',
+  'total_received',
+  'total_fees_sent',
+  'total_miner_rewards_received',
+  'locked',
+];
+
+export const accountBalancesClient = atomFamilyWithQuery<
+  PrincipalWithNetworkUrl,
+  AccountBalanceResponseBigNumber
+>(
+  AccountClientKeys.BalancesClient,
+  async function accountBalancesClientQueryFn(get, [principal, _networkUrl]) {
+    const { accountsApi } = get(apiClientState);
+    const balances = (await accountsApi.getAccountBalance({
+      principal,
+    })) as AddressBalanceResponse;
+    const stx: any = balances.stx;
+    keys.forEach(key => {
+      stx[key] = new BigNumber(balances.stx[key]);
+    });
```

Comment:
Can we add a typing here? Any way to do this less imperatively?

Reply:
Yeah I wasn't sure, open to any suggestions 🙏